core/feature_checker.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+use std::collections::BTreeMap;
+
+#[derive(Default, Debug)]
+pub struct FeatureChecker {
+  features: BTreeMap<&'static str, bool>,
```

Comment:
Maybe `enabled_features: HashSet<&'static str>`

Reply:
Probably a good idea, it's not like there's gonna be entry set to `false` anyway